_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,865 @@
+
+## Install Spinnaker on Google Kubernetes Engine
+
+Spinnaker supports the following installation environments:
+- Distributed installation on Kubernetes
+- Local installation of Debian packages
+- Local git installation from GitHub
+
+For this example, the distributed alternative was selected. But before the installation, learn about some key concepts in Spinnaker configuration.
+
+### Understand Spinnaker Artifacts and Accounts
+
+A _Spinnaker artifact_ is a named JSON object that refers to an external resource, for example, a docker image or a file stored in GitHub. In a pipeline trigger, you can specify an _expected artifact_, and Spinnaker will compare the incoming artifact from the trigger and bind it to be used by the trigger or another stage in the pipeline.
+Also in Kubernetes, the deployed manifests can be provided statically or as an artifact. In this example, manifests are provided as artifacts stored in GitHub.
+
+For Spinnaker to access and act on resources, like docker registries, cloud providers, and code repositories, different types of accounts must be enabled in Spinnaker configuration, along with the credentials. In addition, the configuration has the optional step of associating Spinnaker with a Kubernetes service account, allowing to restrict permissions over the cluster resources granted to Spinnaker.
+
+### Install Halyard
+
+As described on Spinnaker [docs](https://spinnaker.io/docs/setup/install/), the first step is to install [Halyard](https://spinnaker.io/docs/setup/install/halyard/). For a local install in MacOS:
+
+```shell
+curl -O https://raw.githubusercontent.com/spinnaker/halyard/master/install/macos/InstallHalyard.sh
+sudo bash InstallHalyard.sh
+```
+Verify the installation with:
+
+```shell
+hal -v
+```
+
+You must also install [`kubectl`](https://kubernetes.io/docs/tasks/tools/), the Kubernetes command line too, to run commands against the clusters.
+
+### Choose GKE for Spinnaker deployment
+
+The second step is to choose a cloud provider for the environment in which you will install Spinnaker. For Kubernetes, Spinnaker needs a `kubeconfig` file, to access and manage the cluster. For creating a `kubeconfig` for a GKE cluster, you must first create the cluster. GoogleCloud provides a [free tier](https://cloud.google.com/free) of their services that grants you $300 in free credits if you are a new user.
+
+After you signed up, install [`gcloud` CLI](https://cloud.google.com/sdk/docs/install). Following the process to the end, the last step is to run `gcloud init` and set up the authorization for the tool.
+
+Create the cluster for the Spinnaker deployment with the following line:
+```shell
+gcloud container clusters create spinnaker-cluster \
+--zone southamerica-east1-a \
+--machine-type n1-standard-4 \
+--enable-autorepair \
+--enable-autoupgrade
+```
+Then fetch the cluster credentials with:
+
+```shell
+gcloud container clusters get-credentials spinnaker-cluster --zone southamerica-east1-a
+```
```

Comment:
I got this error when I ran this command (after updating the GCloud CLI):

```
CRITICAL: ACTION REQUIRED: gke-gcloud-auth-plugin, which is needed for continued use of kubectl, 
was not found or is not executable. Install gke-gcloud-auth-plugin for use with kubectl by following 
https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke
```

Reply:
I had the suggestion on my list, so I added a note.